### PR TITLE
Update to 1.18

### PR DESCRIPTION
**Pull request by JRoy**
It compiles, it runs. There is probably a lot of location related y-level bugs that I wasn't able to find issues but I also only tested for 10 minutes. @jpenilla said he may do something for mappings stuff

Do not merge until 1.18 is fully released

### CODE REVIEW

**Comment by JRoy** [comment]
Use rebase when merging this

File: Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java [hunk 1]
```diff
@@ -248,12 +257,12 @@ public boolean equals(final Object o) {
                 minor == that.minor &&
                 patch == that.patch &&
                 revision == that.revision &&
-                prerelease == that.prerelease;
+                    preRelease == that.preRelease;
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java]: no

**Review by mdcfe — CHANGES_REQUESTED** [review]

File: Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java (hunk 1 above)
Comment (Bobcat00) [reply Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java:260]: This thing doesn't work with Spigot 1.18-rc3-R0.1-SNAPSHOT

**Comment by Bobcat00** [comment]
This thing doesn't work with Spigot 1.18-rc3-R0.1-SNAPSHOT

**Review by mdcfe — DISMISSED** [review]

**Review by mdcfe — APPROVED** [review]

File: Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java [hunk 2]
```diff
@@ -51,7 +52,8 @@
         builder.put("org.bukkit.Bukkit#getLastTickTime", SupportStatus.DANGEROUS_FORK);
         builder.put("brand:Yatopia", SupportStatus.DANGEROUS_FORK);
         // Yatopia downstream(s) which attempt to do tricky things :)
-        builder.put("brand:Hyalus", SupportStatus.DANGEROUS_FORK);
+        builder.put("brand:" +
+                "Hyalus", SupportStatus.DANGEROUS_FORK);
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java]:
```suggestion
        builder.put("brand:Hyalus", SupportStatus.DANGEROUS_FORK);
```

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Review by mdcfe — APPROVED** [review]

**Pull request merged**